### PR TITLE
Add 'Bucket.labels' property.

### LINKED ISSUE #3473: Add support for bucket labels
See: https://cloud.google.com/storage/docs/json_api/v1/buckets#labels

### PR DESCRIPTION
**Pull request by tseaver**
See: https://cloud.google.com/storage/docs/json_api/v1/buckets#labels

Closes #3473.

### CODE REVIEW

File: storage/google/cloud/storage/bucket.py [hunk 1]
```diff
@@ -552,6 +552,36 @@ def cors(self, entries):
...
+    def labels(self):
+        """Retrieve or set CORS policies configured for this bucket.
+
+        See
+        https://cloud.google.com/storage/docs/json_api/v1/buckets#labels
+
+        :setter: Set labels for this bucket.
+        :getter: Gets the labels for this bucket.
+
+        :rtype: :class:`dict`
```
Comment (dhermes) [inline storage/google/cloud/storage/bucket.py]: Just `dict` would be OK (feel free to leave as-is)

File: storage/tests/unit/test_blob.py [hunk 2]
```diff
@@ -55,7 +55,8 @@ def test_ctor_with_encoded_unicode(self):
         blob_name = b'wet \xe2\x9b\xb5'
         blob = self._make_one(blob_name, bucket=None)
         unicode_name = u'wet \N{sailboat}'
-        self.assertNotEqual(blob.name, blob_name)
+        self.assertNotIsInstance(blob.name, bytes)
```
Comment (dhermes) [inline storage/tests/unit/test_blob.py]: This is Py2/Py3 proof?

File: storage/tests/unit/test_blob.py [hunk 3]
```diff
@@ -55,7 +55,8 @@ def test_ctor_with_encoded_unicode(self):
...
+        self.assertIsInstance(blob.name, type(unicode_name))
```
Comment (dhermes) [inline storage/tests/unit/test_blob.py]: Use `six.text_type`?

File: storage/tests/unit/test_bucket.py [hunk 4]
```diff
@@ -663,6 +666,26 @@ def test_cors_setter(self):
...
+
+    def test_labels_setter(self):
+        NAME = 'name'
+        LABELS = {'color': 'red', 'flavor': 'cherry'}
+        bucket = self._make_one(name=NAME)
+
+        self.assertEqual(bucket.labels, {})
+        bucket.labels = LABELS
+        self.assertEqual(bucket.labels, LABELS)
+        self.assertTrue('labels' in bucket._changes)
```
Comment (dhermes) [inline storage/tests/unit/test_bucket.py]: 1. Use `assertIn('labels', bucket._changes)`?
1. Add `assertIsNot(bucket._properties['labels'], LABELS)`?

File: storage/tests/unit/test_blob.py (hunk 2 above)
Comment (tseaver) [inline storage/tests/unit/test_blob.py]: Yes, we are guaranteed to have text-only names.  The older test (for inequality with the bytes source) was generating a `UnicodeWarning`, which `py.test` just started nannying about.

Comment (dhermes) [inline storage/tests/unit/test_blob.py]: "text-only" is a bit undefined at the Py2/Py3 boundary

File: storage/tests/unit/test_blob.py (hunk 3 above)
Comment (tseaver) [inline storage/tests/unit/test_blob.py]: 32f811b3d06ffd907a3106a4b8133cbd2a685026

File: storage/tests/unit/test_bucket.py (hunk 4 above)
Comment (tseaver) [inline storage/tests/unit/test_bucket.py]: 22bc77c280123c8e5d51478a77ec5cad01a6442d

**Comment by tseaver** [comment]
I will merge when CI is green.

File: storage/tests/unit/test_blob.py (hunk 2 above)
Comment (tseaver) [inline storage/tests/unit/test_blob.py]: On the contrary:  'text' is the *only* way to be precise when straddling.

Comment (dhermes) [inline storage/tests/unit/test_blob.py]: I was under the impression that "text" means `str` (i.e. what you get when `a = 'b'`), but `str == bytes` in Py2 so this test wouldn't succeed there.

File: storage/tests/unit/test_bucket.py [hunk 5]
```diff
@@ -663,6 +666,27 @@ def test_cors_setter(self):
...
+
+    def test_labels_setter(self):
+        NAME = 'name'
+        LABELS = {'color': 'red', 'flavor': 'cherry'}
+        bucket = self._make_one(name=NAME)
+
+        self.assertEqual(bucket.labels, {})
+        bucket.labels = LABELS
+        self.assertEqual(bucket.labels, LABELS)
+        self.assertIsNot(bucket.labels, LABELS)
```
Comment (dhermes) [inline storage/tests/unit/test_bucket.py]: @tseaver I really meant

```python
self.assertIsNot(bucket._properties['labels'], LABELS)
```

In it's current form, the getter indirection doesn't test the same concept

Comment (tseaver) [inline storage/tests/unit/test_bucket.py]: 49df7c3

**Pull request merged**